engines/cge2/talk.cpp

```diff
@@ -254,6 +261,14 @@ InfoLine::InfoLine(CGE2Engine *vm, uint16 w, ColorBank color)
 }
 
 void InfoLine::update(const char *text) {
+	Common::TextToSpeechManager *ttsMan = g_system->getTextToSpeechManager();
+	if (text) {
+        if (strcmp(lastText,text) != 0) {
+            if (ttsMan != nullptr && ConfMan.getBool("tts_enabled"))
+                ttsMan->say(text);
+            lastText = text;
+        }
+    }
```

Comment:
Is the `lastText` check needed?
I would think that the default action, which is `INTERRUPT_NO_REPEAT`, would take care of not repeating the text.

Also could you not use `_oldText` instead of `lastText`?
Actually it seems to me you could move the whole block after the
```c++
if (!_realTime && (text == _oldText))
 		return;
```
you have just a few lines below and remove the strcmp check as a result.

Reply:
I just tried both `INTERRUPT_NO_REPEAT` and moving it after `_oldText`. What happens is the object get repeatedly said over if the cursor is not moved.

Reply:
Ah indeed. Using INTERRUPT_NO_REPEAT would prevent to interrupt and start again the string that it is currently speaking, but once it has finished speaking it, it can say it again.

I would still expect the `text == _oldText` check to do the job, but I might be wrong there. It's possible that the pointers are different but the text value is the same. You should still be able to use the _oldText variable without having to add a `lastText` variable.

Reply:
I think I also tried that and it didn't work as we would like it to

Reply:
I will try to get you branch in the next few days to play a bit with this as I don't know why it would not work.

Reply:
I have verified that you don't need the `_lastText` check to prevent repeating the name of objects when the cursor is over an object.

Actually your current implementation is not ideal as when you move the cursor over an object it will speak its name, but then if you move the cursor away from the object, and then a bit later back on the object (without moving it on another object in between) it will not speak the name again. And I think it is wrong. Relying on the check already there with `_oldText` fixes this.

Since you may not have understood properly what I suggested, here is the code I just tested and seems to work properly for me:
```c++
void InfoLine::update(const char *text) {
	if (!_realTime && (text == _oldText))
		return;

	_oldText = text;

	Common::TextToSpeechManager *ttsMan = g_system->getTextToSpeechManager();
	if (text && ttsMan != nullptr && ConfMan.getBool("tts_enabled_objects"))
		ttsMan->say(text);
```

Reply:
It now works for me. Thanks for the suggestion and code!